ModulesGarden

max-mi opened the pull request:
Added:
    - Separate backend and frontend configuration
    - Separate zendesk API accounts
    - All tickets tab
    - Dashboard ticket statistics
    - User selection while ticket creating
    - Order selection while ticket creating
    - SSO authetification
    - Magento customer and finance data with Zendesk end-user profile synchronization

Fixed:
    - Client loose all tickets if change email in magento
    - Ajax test connection
    - UI fixes
    - Tickets created from "Feedback Tab" or "Contact Us" do not show thread in module ticket list
    - Several language improvements

Commit:
ModulesGarden changelog:
Added:
    - Separate backend and frontend configuration
    - Separate zendesk API accounts
    - All tickets tab
    - Dashboard ticket statistics
    - User selection while ticket creating
    - Order selection while ticket creating
    - SSO authetification
    - Magento customer and finance data with Zendesk end-user profile synchronization

Fixed:
    - Client loose all tickets if change email in magento
    - Ajax test connection
    - UI fixes
    - Tickets created from "Feedback Tab" or "Contact Us" do not show thread in module ticket list
    - Several language improvements

jwswj commented on src/app/code/community/Zendesk/Zendesk/Block/Adminhtml/Config/Buttons/Sync.php:
```diff
@@ -0,0 +1,56 @@
+<?php
+/**
+ * Copyright 2013 Zendesk.
```
We've got a license fine in the root of the repo (https://github.com/zendesk/magento_extension/blob/master/LICENSE). I'm happy to not have these at the top of each file going forward.

jwswj commented on src/app/code/community/Zendesk/Zendesk/Block/Adminhtml/Config/Buttons/Sync.php:
```diff
@@ -0,0 +1,56 @@
+    {
+        return 'Token token="' . Mage::helper('zendesk')->getApiToken(false) . '"';
+    }
+}
```
new line at the end of the file is missing

jwswj commented on src/app/design/adminhtml/default/default/template/zendesk/config/button-signup.phtml:
```diff
@@ -17,7 +17,7 @@
 <script type="text/javascript">
     function postToZendesk() {
-        var form = new Element('form', {method: 'post', action: '<?php echo $this->getPostUrl(); ?>'});
+        var form = new Element('form', {method: 'get', action: '<?php echo $this->getPostUrl(); ?>'});
```
Why did this change? I thought we needed a POST call to kick off the next stage of the signup process?

jwswj commented on src/app/design/adminhtml/default/default/template/zendesk/config/button-test-zendesk.phtml:
```diff
@@ -15,6 +15,27 @@
+        });
+    }
+</script>
+<div id="zendesk-api-connection-results"></div>
```
new line at end of file missing here

jwswj commented on src/app/design/adminhtml/default/default/template/zendesk/widget/form/container.phtml:
```diff
@@ -0,0 +1,55 @@
+
+/* * ********************************************************************
+ * Customization Services by ModulesGarden.com
+ * Copyright (c) ModulesGarden, INBS Group Brand, All Rights Reserved 
```
The module is open sourced under the Apache 2.0 license. You can't have your own copyrighted modules in here.

jwswj commented on src/app/design/frontend/base/default/layout/zendesk.xml:
```diff
@@ -49,5 +49,7 @@
+    <zendesk_sso_login>
+    </zendesk_sso_login>
 
 </layout>
```
new line at end of file

jwswj commented on src/app/code/community/Zendesk/Zendesk/sql/zendesk_setup/mysql4-install-1.3.1.php:
```diff
@@ -0,0 +1,39 @@
+  `admin_user_id` int(10) unsigned NOT NULL COMMENT 'Admin ID',
+  `use_global_settings` tinyint(1) DEFAULT '0' COMMENT 'User use setting from global configuration',
+  `username` varchar(255) DEFAULT NULL COMMENT 'Username',
+  `password` varchar(255) DEFAULT NULL COMMENT 'Password',
```
I'm concerned about zendesk plain text passwords being stored in Magento databases. I'd prefer that this feature was a different Pull Request which we could debate about if we want the feature in the extension or not rather than bundled together with all these other bug fixes and feature improvements.

jwswj commented on src/app/code/community/Zendesk/Zendesk/etc/system.xml:
```diff
@@ -256,18 +249,66 @@
+                <backend_features>
+                    <expanded>1</expanded>
+                    <label>Admin Backend</label>
+                    <comment>Decide which features you would like turned on in your admin panel.</comment>
```
Need a translation for this

jwswj commented on src/app/code/community/Zendesk/Zendesk/etc/system.xml:
```diff
@@ -90,6 +90,26 @@
                             <comment><![CDATA[Depending on your Zendesk configuration under "Settings / Security / JSON Web Token / Update of external IDs?" ]]></comment>
                         </use_external_id>
+                        <customer_sync>
+                            <label>Synchronize Customers</label>
```
Translation required

Commit:
Translations, License,

Commit:
Whole Base

Commit:
Merged master+dashboard

Commit:
signup method

Commit:
Magento Dashboard Grid Fix
Added Default Sort/Dir Options
Added Side Ticket Statistics on Zendesk Dashboard
Statistics enable/disable options
Version changed
Magento Integration App strip header bug fixed

jwswj commented on src/app/code/community/Zendesk/Zendesk/Block/Adminhtml/Create/Order/Grid.php:
```diff
@@ -0,0 +1,137 @@
+
+    public function getGridUrl()
+    {
+                return $this->getUrl('adminhtml/zendesk/loadBlock', array('block'=>'order_grid', '_current' => true, 'collapse' => null));
```
indentation is out here

jwswj commented on src/app/code/community/Zendesk/Zendesk/Block/Adminhtml/Dashboard/Tab/Tickets/Grid/Massaction.php:
```diff
@@ -0,0 +1,32 @@
+class Zendesk_Zendesk_Block_Adminhtml_Dashboard_Tab_Tickets_Grid_Massaction extends Mage_Adminhtml_Block_Widget_Grid_Massaction {
+
+    public function getJavaScript() {
+        return " {$this->getJsObjectName()} = new varienGridMassaction('{$this->getHtmlId()}', "
```
This looks like mess - what's it doing?

max-mi replied:
Native magento grids do not support massactions. Such method appends required js to initialize that feature. 
http://prntscr.com/667oov

jwswj commented on src/app/code/community/Zendesk/Zendesk/Block/Customer/Tickets.php:
```diff
@@ -22,4 +22,8 @@ public function __construct()
     }
+    
+    public function getSubmitAction() {
+        return "http://".Mage::getStoreConfig('zendesk/general/domain')."/anonymous_requests/new";
```
Why use the anonymous_request endpoint rather than the tickets endpoint?

jwswj commented on src/app/code/community/Zendesk/Zendesk/Block/Customer/Tickets.php:
```diff
@@ -22,4 +22,8 @@ public function __construct()
+    public function getSubmitAction() {
+        return "http://".Mage::getStoreConfig('zendesk/general/domain')."/anonymous_requests/new";
+    }
 }
```
Line and end of file missing here

jwswj commented on src/app/code/community/Zendesk/Zendesk/Block/Supporttab.php:
```diff
@@ -19,10 +19,10 @@ class Zendesk_Zendesk_Block_Supporttab extends Mage_Core_Block_Template
-        return Mage::getStoreConfig('zendesk/features/feedback_tab_code');
+        return Mage::getStoreConfig('zendesk/frontend_features/feedback_tab_code');
     }
 }
```
line at end of file missing here too

jwswj commented on src/app/code/community/Zendesk/Zendesk/Helper/Data.php:
```diff
@@ -188,4 +268,230 @@ public function isExternalIdEnabled()
+        $totals = array(
+            'open'      =>  0,
+            'new'       =>  0,
+            'solved'    =>  0,
```
There is also an "on-hold" status which can be enabled for accounts. Is there a way these can be not hard-coded? Seem brittle.

jwswj commented on src/app/code/community/Zendesk/Zendesk/Helper/Data.php:
```diff
@@ -188,4 +268,230 @@ public function isExternalIdEnabled()
+    public function getTicketTotals($type = null, $from = null, $to = null)
+    {
+        try {
+        $tickets = Mage::getModel('zendesk/api_tickets')->all();
```
Indentation is off

jwswj commented on src/app/code/community/Zendesk/Zendesk/Model/Api/Tickets.php:
```diff
@@ -87,6 +87,17 @@ public function recent()
+    public function all()
+    {
+        $response = $this->_call('tickets.json');
+        return $response['tickets'];
```
Does this handle pagination? What about if an account has 10000s of tickets?

max-mi replied:
It was used to calculate ticket statistics

jwswj commented on src/app/code/community/Zendesk/Zendesk/Helper/Data.php:
```diff
@@ -188,4 +268,230 @@ public function isExternalIdEnabled()
+        else
+            $from = 0;
+
+        foreach( $tickets as $ticket )
```
I don't want stats calculation to be completed in the Magento extension. Stats should be calculated in the Zendesk system, then just displayed here. Unfortunately there isn't a V2 stats API so this functionality will just have to be removed for now.

jwswj commented on src/app/design/frontend/base/default/template/zendesk/customer/tickets.phtml:
```diff
@@ -28,4 +28,7 @@
+</div>
+<div class="buttons-set">
+    <a class="button" target="_blank" href="<?php echo Mage::helper('zendesk')->getSSOAuthUrlEndUsers(); ?>"><span><span><?php echo $this->__('Log In To Zendesk') ?></span></span></a>
 </div>
```
new line at eof

jwswj commented on src/skin/adminhtml/default/default/zendesk/js/validation.js:
```diff
@@ -0,0 +1,23 @@
+            $rows.toggle( $(this).val() === '0');
+        }).trigger('change');
+    });
+})($jq);
```
new line at eof

jwswj commented on src/app/design/adminhtml/default/default/template/zendesk/widget/grid.phtml:
```diff
@@ -0,0 +1,227 @@
+//]]>
+</script>
+<?php endif; ?>
+<?php endif ?>
```
new line and eof and also missing a ; for consistency with previous line

jwswj commented on src/app/design/adminhtml/default/default/template/zendesk/left-menu.phtml:
```diff
@@ -41,8 +41,34 @@
+    </div>
+</div>
+<?php endif; ?>
+<?php endif; ?>
```
new line at eof missing

jwswj commented on src/app/design/adminhtml/default/default/template/zendesk/dashboard/index.phtml:
```diff
@@ -41,4 +135,5 @@
-<?php endif; ?>
\ No newline at end of file
+<?php endif; ?>
+<?php endif ?>
```
missing `;`

jwswj commented on src/app/design/adminhtml/default/default/template/zendesk/dashboard/index.phtml:
```diff
@@ -15,8 +15,102 @@
+    document.getElementById('tickets-all').innerHTML = values.all;
+    document.getElementById('tickets-pending').innerHTML = values.pending;
+}
+function checkStatistics() {
```
This statistics logic is just too brittle to launch. Need to wait for stats API.

jwswj commented on src/app/design/adminhtml/default/default/template/zendesk/dashboard/index.phtml:
```diff
@@ -15,8 +15,102 @@
+            method: 'get',
+            parameters: {from: document.getElementById('ticket_from').value, to: document.getElementById('ticket_to').value},
+            requestHeaders: {
+                Authorization: '<?php echo $this->getAuthHeader(); ?>'
```
Auth header should never be rendered into the frontend.

jwswj commented on src/app/design/adminhtml/default/default/template/zendesk/create/order.phtml:
```diff
@@ -0,0 +1,45 @@
+<script>
+document.getElementById('zendesk_create_order_search_grid').parentNode.style.display = 'none';
+function insertOrder(id) {
+        var request = new Ajax.Request('<?php echo $this->getUrl('adminhtml/zendesk/getOrder'); ?>'+'id/'+id, {
```
indentation is different to lines below

jwswj commented on src/app/design/adminhtml/default/default/template/zendesk/create/order.phtml:
```diff
@@ -0,0 +1,45 @@
+        });
+}
+function showOrders() {
+    document.getElementById('zendesk_create_order_search_grid').parentNode.style.display = 'block';
```
Can this be done with CSS rather than JS? I'd prefer that classes are used for hiding and displaying rather than manipulating styles inline on the element.

jwswj commented on src/app/design/adminhtml/default/default/template/zendesk/create/customer.phtml:
```diff
@@ -0,0 +1,46 @@
+?>
+<div class="entry-edit">
+    <div class="entry-edit-head">
+        <div style="float: right;"><?php echo $this->getButtonsHtml() ?></div>
```
style shouldn't be inline here

jwswj commented on src/app/design/adminhtml/default/default/template/zendesk/config/button-sync.phtml:
```diff
@@ -0,0 +1,42 @@
+<button style="" onclick="javascript:sync(); return false;" class="scalable" type="button" id="<?php echo $this->getHtmlId() ?>">
+    <span><?php echo $this->escapeHtml($this->getButtonLabel()) ?></span>
+</button>
+<div id="zendesk-sync-results"></div>
```
missing line at eof

max-mi replied:
It seems that github cut it automatically

jwswj commented on src/app/code/community/Zendesk/Zendesk/etc/config.xml:
```diff
@@ -19,18 +19,28 @@
     </modules>
     <zendesk>
-        <provision_url>https://signup.zendesk.com/provisioning/magento/welcome</provision_url>
+        <provision_url>https://support.zendesk.com/hc/en-us/articles/203660046-Magento-Integration-Installing-the-Zendesk-extension-for-Magento</provision_url>
```
Why has this URL changed? It should still be pointing to the provisioning service

jwswj commented on src/app/code/community/Zendesk/Zendesk/controllers/Adminhtml/ZendeskController.php:
```diff
@@ -346,18 +422,433 @@ public function clearLogAction()
+        }
+        catch (Exception $ex) 
+        {
+            $this->getResponse()->setBody(json_encode(array('success'=>false, 'msg'=>Mage::helper('zendesk')->__('Synchronization failed'))));
```
Can you add some more logging here about why it failed?

jwswj commented on src/app/code/community/Zendesk/Zendesk/Model/Source/Sortdir.php:
```diff
@@ -0,0 +1,37 @@
+ * limitations under the License.
+ */
+
+class Zendesk_Zendesk_Model_Source_Sortdir {
```
Is this really required? Can you explain what it does?

max-mi replied:
It is used for configuration select menu.
http://prntscr.com/697vh7

jwswj commented on src/app/code/community/Zendesk/Zendesk/Model/Search/Field.php:
```diff
@@ -0,0 +1,50 @@
+        return $this->getString();
+    }
+    
+}
```
Missing line at EOF

jwswj commented on src/app/code/community/Zendesk/Zendesk/Model/Resource/Tickets/Collection.php:
```diff
@@ -0,0 +1,149 @@
+        if(is_string($condition) OR is_array($condition)) {
+            
+            switch($fieldName) {
+                case 'subject':
```
There might be more of these fields in the future? What functionality is this used for? It seems brittle and not needed at the moment.

max-mi replied:
It is used for filtering and searching by this fields in Magento Grid:
http://prntscr.com/667pfs

jwswj replied:
This conditional would be a lot cleaner if it only called `$this->_search->addField( new Zendesk_Zendesk_Model_Search_Field( ));` once at the end of the function. What about if you extracted this into 2 logical blocks: build the search field value, then add it to the field. Right now it's very messy.

jwswj commented on src/app/code/community/Zendesk/Zendesk/Model/Api/Abstract.php:
```diff
@@ -95,15 +99,24 @@ protected function _call(
         return $body;
     }
-}
+}
```
missing line at end of file

jwswj commented on src/app/code/community/Zendesk/Zendesk/Helper/Data.php:
```diff
@@ -188,4 +268,230 @@ public function isExternalIdEnabled()
+        return array(
+            'new'       =>  'New',
+            'open'      =>  'Open',
+            'pending'   =>  'Pending',
```
on-hold status also exists

jwswj commented on src/app/code/community/Zendesk/Zendesk/Helper/Data.php:
```diff
@@ -188,4 +268,230 @@ public function isExternalIdEnabled()
+        }
+    }
+    
+    public function getExcerpt($row)
```
Please just use the subject of a ticket. We don't need logic in the Magento extension which tries to work out an Excerpt

max-mi replied:
getExcerpt function returns ticket subject if it exists. Function was implemented because of tickets which have no subject (they were not listed).

jwswj replied:
It's very uncommon for a ticket to not have a subject. I'd prefer that it was blank.

max-mi replied:
Tickets which are created from contact us form don't have subject. Title is a direct link to the ticket in zendesk, if it will be blank there will not be any way to go to this ticket. Do you really want to leave it blank or maybe set to "No Subject"?

jwswj replied:
Use no subject please

jwswj commented:
@max-mi there are a huge amount of changes in here. This Pull Request is just getting too big to really understand and digest. Can you please not add anymore features to it. Please remove the custom agent authentication as previously request, also the stats calculations which you have added, then clean up all the things I have commented on.

Commit:
Bulk Action Error Handling,
Tickets Counters Fix,
Formatting, New Lines

Commit:
Authorization when viewing ticket
Added Type Column
Email filtering Fix

Commit:
Sale Totals Synchronization Fix

Commit:
Sync Logging
Inline CSS -> JS

Commit:
Customer Open Ticket SSO

Commit:
Stats Removed
Commented things fix

Commit:
SSO Disabled Fix, Create Ticket Blank Page Fix

Commit:
Massaction Views Fix

Commit:
Upgrade 1.3.0/1-1.4.0

Commit:
Dashboard Grid Type Dropdown Filter

Commit:
Subject Filtering Fix

jwswj commented on src/app/code/community/Zendesk/Zendesk/Model/Search.php:
```diff
@@ -0,0 +1,50 @@
+    public function getString() {
+        return 'type:' . $this->type . $this->separator . implode($this->separator, $this->fields);
+    }
+}
```
new line at EOF missing here

jwswj commented on src/app/code/community/Zendesk/Zendesk/Model/Resource/Tickets/Collection.php:
```diff
@@ -0,0 +1,158 @@
+            
+            $this->appendParamsWithoutIdPostfix($ticket, array('requester', 'assignee', 'group'));
+            
+            $obj    = new Varien_Object();
```
spacing seems off here

jwswj commented:
The placement of `{` after functions and if statements is inconsistent in these changes. Please keep the opening `{` on the same line as the function or if statement.

jwswj commented on src/app/code/community/Zendesk/Zendesk/controllers/Adminhtml/ZendeskController.php:
```diff
@@ -80,9 +98,35 @@ public function authenticateAction()
+            }
+            catch( Exception $exc )
+            {
+                //just do nothing
```
why do nothing?

jwswj commented on src/app/code/community/Zendesk/Zendesk/controllers/Adminhtml/ZendeskController.php:
```diff
@@ -346,18 +435,444 @@ public function clearLogAction()
+        Mage::register('zendesk_groups', $groups);
+    }
+    
+    protected function getMassactionResponse($response, $ids, $message = '%d out of %d ticket(s) were updated.')
```
should the function name be `getMassActionResponse`?

jwswj commented on src/app/code/community/Zendesk/Zendesk/controllers/Adminhtml/ZendeskController.php:
```diff
@@ -346,18 +435,444 @@ public function clearLogAction()
+            }
+
+            Mage::getSingleton('adminhtml/session')->addSuccess(
+                    Mage::helper('zendesk')->__(
```
indentation looks off here

jwswj commented on src/skin/adminhtml/default/default/zendesk/zendesk.css:
```diff
@@ -45,4 +45,37 @@ h3.zendesk-header {
+}
+#zendesk_tab_tickets_grid_all_table .zendesk-grid-tr:hover {
+    background: #fcf5dd;
 }
```
missing line at eof

jwswj commented on src/app/code/community/Zendesk/Zendesk/sql/zendesk_setup/mysql4-install-1.4.0.php:
```diff
@@ -0,0 +1,30 @@
+
+$installer = $this;
+$installer->startSetup();
+$installer->run("CREATE TABLE IF NOT EXISTS `{$installer->getTable('zendesk_settings')}` (
```
Are these tables still required as this PR no longer includes the individual user settings?

jwswj commented:
@max-mi this is looking really good - just a few style final nit-picks which I've commented on, then I'd be happy to merge.

Commit:
Config API Views Blank Page Fix,
Migrate Settings 1.3.0/1-1.4.0,
SSO Tickets Links From Customer View,
SSO Ticket Link When Ticket is Created

Commit:
UI Grid Fixes

Commit:
Indentation, EOF Fixes

Commit:
Reformat

Commit:
Magento Exception Handling,
Refactor addFieldToFilter method

Commit:
Separate User Settings

Commit:
Separate User Settings

Commit:
User Settings moved on other branch

jwswj commented on src/app/code/community/Zendesk/Zendesk/controllers/Adminhtml/ZendeskController.php:
```diff
@@ -129,11 +154,25 @@ public function logoutAction()
+                $this->_redirect('adminhtml/zendesk/index');
+                return;
+            }
+        } catch( Exception $ex ) {
```
When would this exception be thrown? Seems like we're being too defensive here and also throwing exactly the same error wouldn't help with debugging.

jwswj commented on src/app/code/community/Zendesk/Zendesk/controllers/Adminhtml/ZendeskController.php:
```diff
@@ -162,11 +201,29 @@ public function createAction()
     {
-        $this->_redirectUrl(Mage::helper('zendesk')->getUrl());
+        $domain = Mage::getStoreConfig('zendesk/general/domain');
+        if(!$domain) {
```
Can this logic be extracted into a private method?

jwswj commented on src/app/code/community/Zendesk/Zendesk/controllers/Adminhtml/ZendeskController.php:
```diff
@@ -346,18 +403,317 @@ public function clearLogAction()
+                )
+            );
+            $data[] = array(
+                'user_field' => array(
```
Why sync customer name into it's own field, rather than into the default user field?

max-mi replied:
It also updates default user field, should I remove this one?

jwswj commented on src/app/code/community/Zendesk/Zendesk/Model/Observer.php:
```diff
@@ -109,4 +110,97 @@ public function addTicketButton(Varien_Event_Observer $event)
+                    "group"         =>  $group->getCode(),
+                    "name"          =>  $customer->getFirstname() . " " . $customer->getLastname(),
+                    "id"            =>  $customer->getId(),
+                    "logged_in"     =>  date("Y-m-d\TH:i:s\Z",strtotime($log_customer->getLoginAt())),
```
If a customer hasn't logged in before this returns `1970-01-01T00:00:00Z`. It should just be empty if the customer hasn't logged in.

jwswj commented on src/app/code/community/Zendesk/Zendesk/Model/Observer.php:
```diff
@@ -109,4 +110,97 @@ public function addTicketButton(Varien_Event_Observer $event)
+                "user_fields"       =>  array(
+                    "group"         =>  $group->getCode(),
+                    "name"          =>  $customer->getFirstname() . " " . $customer->getLastname(),
+                    "id"            =>  $customer->getId(),
```
If `isExternalIdEnabled` then shouldn't this also set external_id on the user?

max-mi replied:
$customer->getId() returns external id for it.

jwswj commented on src/app/design/adminhtml/default/default/template/zendesk/dashboard/index.phtml:
```diff
@@ -29,16 +30,17 @@
                     <span><?php echo $this->__('View All Tickets'); ?></span>
                 </button>
             </div>
         <?php endif; ?>
```
indentation is wrong here

jwswj commented on src/app/design/adminhtml/default/default/template/zendesk/dashboard/index.phtml:
```diff
@@ -29,16 +30,17 @@
     <?php echo $this->getChildHtml('zendesk_dashboard_empty'); ?>
-<?php endif; ?>
\ No newline at end of file
+<?php endif; ?>
```
indentation is wrong here too

jwswj commented on src/app/design/adminhtml/default/default/template/zendesk/tickets/tickets.phtml:
```diff
@@ -0,0 +1,74 @@
+    } catch(Exception $e) {
+        // Don't do anything, just don't show the tickets
+    }
+
```
extra line isn't required

jwswj commented on src/app/design/adminhtml/default/default/template/zendesk/widget/grid.phtml:
```diff
@@ -0,0 +1,227 @@
+</div>
+<script type="text/javascript">
+//<![CDATA[
+    <?php echo $this->getJsObjectName() ?> = new varienGrid('<?php echo $this->getId() ?>', '<?php echo $this->getGridUrl() ?>', '<?php echo $this->getVarNamePage() ?>', '<?php echo $this->getVarNameSort() ?>', '<?php echo $this->getVarNameDir() ?>', '<?php echo $this->getVarNameFilter() ?>');
```
This is just too messy and it seems to be a php version of https://github.com/max-mi/magento_extension/blob/master/src/app/code/community/Zendesk/Zendesk/Block/Adminhtml/Dashboard/Tab/Tickets/Grid/Massaction.php#L22-L30
Can it be cleaned up or extracted at all?

max-mi replied:
I've rewritten this into PHP.

jwswj commented on src/app/design/adminhtml/default/default/template/zendesk/widget/grid.phtml:
```diff
@@ -0,0 +1,227 @@
+        </tr>
+    </table>
+<?php endif; ?>
+<?php if($this->getMassactionBlock()->isAvailable()): ?>
```
why not just:

``` php
<?php
if($this->getMassactionBlock()->isAvailable()){
  echo $this->getMassactionBlockHtml();
}
?>
```

jwswj commented on src/skin/adminhtml/default/default/zendesk/js/validation.js:
```diff
@@ -0,0 +1,23 @@
+Validation.add('validate-require-if-not-global', 'This field is required if you want to use your own configuration.', function(the_field_value){
```
Does this string get internationalised?

max-mi replied:
Script was required in Magento/Zendesk user settings separating.

jwswj commented on src/app/design/adminhtml/default/default/template/zendesk/widget/grid.phtml:
```diff
@@ -0,0 +1,227 @@
+
+            <?php if ($this->shouldRenderSubTotal($_item)): ?>
+                <tr class="subtotals">
+                    <?php $i = 0; foreach ($this->getSubTotalColumns() as $_column): ?>
```
the spacing in this foreach is inconsistent with other similar loops in this file.

Commit:
Refactoring

Commit:
Sync: default last log in date

jwswj commented:
:+1:
